cypress/e2e/widgets/widget-pip-close.spec.ts

```diff
@@ -0,0 +1,205 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { IWidget } from "matrix-widget-api/src/interfaces/IWidget";
+// import { RoomStateEvent } from "matrix-js-sdk/src/matrix";
+
+import type { MatrixClient, MatrixEvent } from "matrix-js-sdk/src/matrix";
+import { SynapseInstance } from "../../plugins/synapsedocker";
+import { UserCredentials } from "../../support/login";
+
+const DEMO_WIDGET_ID = "demo-widget-id";
+const DEMO_WIDGET_NAME = "Demo Widget";
+const DEMO_WIDGET_TYPE = "demo";
+const ROOM_NAME = "Demo";
+
+const DEMO_WIDGET_HTML = `
+    <html lang="en">
+        <head>
+            <title>Demo Widget</title>
+            <script>
+                window.onmessage = ev => {
+                    if (ev.data.action === 'capabilities') {
+                        window.parent.postMessage(Object.assign({
+                            response: {
+                                capabilities: []
+                            },
+                        }, ev.data), '*');
+                    }
+                };
+            </script>
+        </head>
+        <body>
+            <button id="demo">Demo</button>
+        </body>
+    </html>
+`;
+
+// mostly copied from src/utils/WidgetUtils.waitForRoomWidget with small modifications
+function waitForRoomWidget(matrixClient: MatrixClient, widgetId: string, roomId: string, add: boolean): Promise<void> {
+    return new Promise((resolve, reject) => {
+        function eventsInIntendedState(evList) {
+            const widgetPresent = evList.some((ev) => {
+                return ev.getContent() && ev.getContent()['id'] === widgetId;
+            });
+            if (add) {
+                return widgetPresent;
+            } else {
+                return !widgetPresent;
+            }
+        }
+
+        const room = matrixClient.getRoom(roomId);
+
+        const startingWidgetEvents = room.currentState.getStateEvents('im.vector.modular.widgets');
+        if (eventsInIntendedState(startingWidgetEvents)) {
+            resolve();
+            return;
+        }
+
+        function onRoomStateEvents(ev: MatrixEvent) {
+            if (ev.getRoomId() !== roomId || ev.getType() !== "im.vector.modular.widgets") return;
+
+            const currentWidgetEvents = room.currentState.getStateEvents('im.vector.modular.widgets');
+
+            if (eventsInIntendedState(currentWidgetEvents)) {
+                // @ts-ignore
+                matrixClient.removeListener('RoomState.events', onRoomStateEvents);
```

Comment:
RoomStateEvents should be accessible via `matrixcs` on the AUTWindow

Reply:
thanks for hint, it is used now